server/src/config.rs

```diff
@@ -0,0 +1,225 @@
+use crate::{PersistedDatabaseRules, ProvidedDatabaseRules};
+use async_trait::async_trait;
+use data_types::server_id::ServerId;
+use generated_types::database_rules::encode_persisted_database_rules;
+use generated_types::google::FieldViolation;
+use generated_types::influxdata::iox::management;
+use iox_object_store::IoxObjectStore;
+use object_store::ObjectStore;
+use snafu::{ensure, ResultExt, Snafu};
+use std::sync::Arc;
+use uuid::Uuid;
+
+/// A generic opaque error
+pub type StdError = Box<dyn std::error::Error + Send + Sync + 'static>;
+
+/// A generic trait for interacting with the configuration
+/// of a database server
+#[async_trait]
+pub trait ConfigProvider: std::fmt::Debug + Send + Sync {
+    /// Returns a list of database name and UUID pairs
+    async fn fetch_server_config(
+        &self,
+        server_id: ServerId,
+    ) -> Result<Vec<(String, Uuid)>, StdError>;
+
+    /// Returns a list of database names and UUID pairs
```

Comment:
```suggestion
    /// Persists a list of database names and UUID pairs
```

It also might help to make it clear if this list overwrites the existing list, or if it is additive, or if it only writes deltas, etc